rust/operator/src/kerberos.rs

```diff
@@ -0,0 +1,139 @@
+use stackable_hdfs_crd::{
+    constants::{HADOOP_SECURITY_AUTHENTICATION, SSL_CLIENT_XML, SSL_SERVER_XML},
+    HdfsCluster, HdfsRole, KerberosConfig,
+};
+
+use crate::config::{CoreSiteConfigBuilder, HdfsSiteConfigBuilder};
+
+impl HdfsSiteConfigBuilder {
+    pub fn kerberos_config(&mut self, hdfs: &HdfsCluster) -> &mut Self {
+        if let Some(kerberos_config) = hdfs.kerberos_config() {
+            self.add("dfs.block.access.token.enable", "true")
+                .add("dfs.http.policy", "HTTPS_ONLY")
+                .add("hadoop.kerberos.keytab.login.autorenewal.enabled", "true")
+                .add("dfs.https.server.keystore.resource", SSL_SERVER_XML)
+                .add("dfs.https.client.keystore.resource", SSL_CLIENT_XML);
+            self.add_wire_encryption_settings(kerberos_config);
+        }
+        self
+    }
+
+    pub fn kerberos_discovery_config(&mut self, hdfs: &HdfsCluster) -> &mut Self {
+        if let Some(kerberos_config) = hdfs.kerberos_config() {
+            // We want e.g. hbase to automatically renew the Kerberos tickets.
+            // This shouldn't harm any other consumers.
+            self.add("hadoop.kerberos.keytab.login.autorenewal.enabled", "true");
+            self.add_wire_encryption_settings(kerberos_config);
+        }
+        self
+    }
+
+    fn add_wire_encryption_settings(&mut self, kerberos_config: &KerberosConfig) -> &mut Self {
+        match kerberos_config.wire_encryption {
+            stackable_hdfs_crd::WireEncryption::Authentication => {
+                self.add("dfs.data.transfer.protection", "authentication");
+                self.add("dfs.encrypt.data.transfer", "false");
+            }
+            stackable_hdfs_crd::WireEncryption::Integrity => {
+                self.add("dfs.data.transfer.protection", "integrity");
+                self.add("dfs.encrypt.data.transfer", "false");
+            }
+            stackable_hdfs_crd::WireEncryption::Privacy => {
+                self.add("dfs.data.transfer.protection", "privacy");
+                self.add("dfs.encrypt.data.transfer", "true");
+            }
+        }
+        self
+    }
+}
+
+impl CoreSiteConfigBuilder {
+    pub fn kerberos_config(
+        &mut self,
+        hdfs: &HdfsCluster,
+        role: &HdfsRole,
+        hdfs_name: &str,
+        hdfs_namespace: &str,
+    ) -> &mut Self {
+        if let Some(kerberos_config) = hdfs.kerberos_config() {
+            self
+                .add("hadoop.security.authentication", "kerberos")
+                .add("hadoop.security.authorization", "true")
+                // Otherwise we fail with `java.io.IOException: No groups found for user nn`
+                // Default value is `dr.who=`, so we include that here
+                .add("hadoop.user.group.static.mapping.overrides", "dr.who=;nn=;nm=;jn=;")
```

Comment:
Note to self that this is a placeholder until we have group mapping (https://github.com/stackabletech/opa-operator/pull/433)

Reply:
Closing, as this was a note to self